src/data-fetchers/vcf/vcf-data-fetcher.test.ts

```diff
@@ -0,0 +1,29 @@
+import { recordToTile } from './vcf-data-fetcher';
+import type { VcfRecord } from './vcf-data-fetcher';
+
+describe('VCF file parser', () => {
+    it('Convert a VCF record to a tile data', () => {
+        const record: VcfRecord = {
+            CHROM: '20',
+            POS: 14370,
+            ID: ['rs6054257'],
+            REF: 'G',
+            ALT: ['A'],
+            QUAL: 29,
+            FILTER: 'PASS',
+            INFO: {
+                NS: [3],
+                DP: [14],
+                AF: ['0.5', 'DB', 'H2']
+            }
+        };
+        const chrPos = 123456;
+        const prevPos = record.POS - 100;
+        const tile = recordToTile(record, chrPos, prevPos);
+
+        expect(tile.MUTTYPE).toBe('substitution');
+        expect(tile.SUBTYPE).toBe('C>T');
+        expect(tile.DISTPREV).toBe(100);
+        expect(tile.NS).toBe('3'); // always string
```

Comment:
I think this would be a really good choice to use a [snapshot test](https://vitest.dev/guide/snapshot.html#inline-snapshots). Snapshot testing is especially useful for this kind of thing where you just want to assert that the transformation is "correct", and to prevent regressions in the future. It also means you can write many fewer assertions. 

To create a snapshot, you can replace the above with the following in the source code:

```suggestion
        expect(tile).toMatchInlineSnapshot();
```

Then, run `yarn test`. The very first time it runs, `vitest` will take a "snapshot" of `tile`, and generate a string _inline_. For example, when I run the tests, the following is generated:

```typescript
        expect(tile).toMatchInlineSnapshot(`
          {
            "AF": "0.5, DB, H2",
            "ALT": "A",
            "CHROM": "20",
            "DISTPREV": 100,
            "DISTPREVLOGE": 4.605170185988092,
            "DP": "14",
            "FILTER": "PASS",
            "ID": [
              "rs6054257",
            ],
            "INFO": "{\\"NS\\":[3],\\"DP\\":[14],\\"AF\\":[\\"0.5\\",\\"DB\\",\\"H2\\"]}",
            "MUTTYPE": "substitution",
            "NS": "3",
            "ORIGINALPOS": 14370,
            "POS": 137827,
            "POSEND": 137828,
            "QUAL": 29,
            "REF": "G",
            "SUBTYPE": "C>T",
          }
        `);
```

Once the snapshot is correct. Make a commit checking in the snapshot to repo. In subsequent test runs, `vitest` will assert that `tile` matches this snapshot exactly. This will prevent regressions from whatever we previously checked into the repo.

If you notice that something is wrong or need fixing, it either means we need to 1.) update the snapshot or 2.) we have a regression that needs fixing.

cc: @etowahadams, useful testing feature of vitest that might be good to know

Reply:
This is really cool! Thanks for sharing this.

Reply:
yeah! might be useful for testing a variety of spec transformations